### PR TITLE
[PLT-4843] Add CLI tool for deleting channels

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
 Add CLI tool for permanently deleting channels

#### Ticket Link
JIRA: https://mattermost.atlassian.net/browse/PLT-4843
GitHub: https://github.com/mattermost/platform/issues/6201

### CODE REVIEW

File: cmd/platform/channel.go [hunk 1]
```diff
@@ -226,6 +240,43 @@ func deleteChannelsCmdF(cmd *cobra.Command, args []string) error {
...
+	initDBCommandContextCobra(cmd)
+
+	if len(args) < 1 {
+		return errors.New("Enter at least one channel to delete.")
+	}
+
+	confirmFlag, _ := cmd.Flags().GetBool("confirm")
+	if !confirmFlag {
+		var confirm string
+		CommandPrettyPrintln("Are you sure you want to delete the teams specified?  All data will be permanently deleted? (YES/NO): ")
```
Comment (coreyhulen) [inline cmd/platform/channel.go]: `teams specified` should be `channels specified` ?

Comment (cpanato) [inline cmd/platform/channel.go]: my bad, fixed

File: cmd/platform/channel.go [hunk 2]
```diff
@@ -205,7 +219,7 @@ func addUserToChannel(channel *model.Channel, user *model.User, userArg string)
 	}
 }
 
-func deleteChannelsCmdF(cmd *cobra.Command, args []string) error {
+func archiveChannelsCmdF(cmd *cobra.Command, args []string) error {
```
Comment (JeffSchering) [inline cmd/platform/channel.go]: Hi @cpanato I was just looking at this for the docs, and I noticed that in this function, "delete" is used in a couple of places:
``` golang
	if len(args) < 1 {
		return errors.New("Enter at least one channel to delete.")
	}

	channels := getChannelsFromChannelArgs(args)
	for i, channel := range channels {
		if channel == nil {
			CommandPrintErrorln("Unable to find channel '" + args[i] + "'")
			continue
		}
		if result := <-app.Srv.Store.Channel().Delete(channel.Id, model.GetMillis()); result.Err != nil {
			CommandPrintErrorln("Unable to delete channel '" + channel.Name + "' error: " + result.Err.Error())
		}
	}
```
Should that be changed to "archive"?
Like this:
`return errors.New("Enter at least one channel to archive.")`
and
`CommandPrintErrorln("Unable to archive '" + channel.Name + "' error: " + result.Err.Error())`

Also, I noticed that the archive command seems to be available in all editions, but the restore command is available in Enterprise only. Is that intentional?

Comment (cpanato) [inline cmd/platform/channel.go]: @JeffSchering the wording is wrong in the archive channels I will submit a PR to change that.

for the restore feature is not available to other editions I guess is intentional, but @coreyhulen can confirm that.

**Pull request merged**